config/fs.go

```diff
@@ -0,0 +1,239 @@
+	RequestTimeoutMsec int64
+}
+
+// RetryableHTTPClientConfig is the complete config for a retryable http client
+type RetryableHTTPClientConfig struct {
+	TimeoutConfig
+	RetryConfig
+}
+
+func parseFSConfig(cfg *Config) {
+	if cfg.FuseConfig.AttrTimeout == 0 {
+		cfg.FuseConfig.AttrTimeout = defaultFuseTimeoutSec
+	}
+
+	if cfg.FuseConfig.EntryTimeout == 0 {
+		cfg.FuseConfig.EntryTimeout = defaultFuseTimeoutSec
+	}
+
+	if cfg.FuseConfig.NegativeTimeout == 0 {
+		cfg.FuseConfig.NegativeTimeout = defaultFuseTimeoutSec
+	}
+
+	if cfg.BackgroundFetchConfig.FetchPeriodMsec == 0 {
+		cfg.BackgroundFetchConfig.FetchPeriodMsec = defaultBgFetchPeriodMsec
+	}
+
+	if cfg.BackgroundFetchConfig.SilencePeriodMsec == 0 {
+		cfg.BackgroundFetchConfig.SilencePeriodMsec = defaultBgSilencePeriodMsec
+	}
+
+	if cfg.BackgroundFetchConfig.MaxQueueSize == 0 {
+		cfg.BackgroundFetchConfig.MaxQueueSize = defaultBgMaxQueueSize
+	}
+
+	if cfg.BackgroundFetchConfig.EmitMetricPeriodSec == 0 {
+		cfg.BackgroundFetchConfig.EmitMetricPeriodSec = defaultBgMetricEmitPeriodSec
+	}
+
+	if cfg.MountTimeoutSec == 0 {
+		cfg.MountTimeoutSec = defaultMountTimeoutSec
+	}
+
+	if cfg.FuseMetricsEmitWaitDurationSec == 0 {
+		cfg.FuseMetricsEmitWaitDurationSec = defaultFuseMetricsEmitWaitDurationSec
+	}
+
+	// RetryableHTTPClientConfig
+
+	if cfg.RetryableHTTPClientConfig.TimeoutConfig.DialTimeoutMsec == 0 {
+		cfg.RetryableHTTPClientConfig.TimeoutConfig.DialTimeoutMsec = defaultDialTimeoutMsec
+	}
+
+	if cfg.RetryableHTTPClientConfig.TimeoutConfig.ResponseHeaderTimeoutMsec == 0 {
+		cfg.RetryableHTTPClientConfig.TimeoutConfig.ResponseHeaderTimeoutMsec = defaultResponseHeaderTimeoutMsec
+
+	}
+	if cfg.RetryableHTTPClientConfig.TimeoutConfig.RequestTimeoutMsec == 0 {
+		cfg.RetryableHTTPClientConfig.TimeoutConfig.RequestTimeoutMsec = defaultRequestTimeoutMsec
+
+	}
+	if cfg.RetryableHTTPClientConfig.RetryConfig.MaxRetries == 0 {
+		cfg.RetryableHTTPClientConfig.RetryConfig.MaxRetries = defaultMaxRetries
+
+	}
+	if cfg.RetryableHTTPClientConfig.RetryConfig.MinWaitMsec == 0 {
+		cfg.RetryableHTTPClientConfig.RetryConfig.MinWaitMsec = defaultMinWaitMsec
+
+	}
+	if cfg.RetryableHTTPClientConfig.RetryConfig.MaxWaitMsec == 0 {
+		cfg.RetryableHTTPClientConfig.RetryConfig.MaxWaitMsec = defaultMaxWaitMsec
+	}
+
+	// BlobConfig
+
+	if cfg.BlobConfig.ValidInterval == 0 {
+		cfg.BlobConfig.ValidInterval = defaultValidIntervalSec
+	}
+	if cfg.BlobConfig.CheckAlways {
+		cfg.BlobConfig.ValidInterval = 0
+	}
+	if cfg.BlobConfig.FetchTimeoutSec == 0 {
+		cfg.BlobConfig.FetchTimeoutSec = defaultFetchTimeoutSec
+	}
+	if cfg.BlobConfig.MaxRetries == 0 {
+		cfg.BlobConfig.MaxRetries = cfg.RetryableHTTPClientConfig.RetryConfig.MaxRetries
+	}
+	if cfg.BlobConfig.MinWaitMsec == 0 {
+		cfg.BlobConfig.MinWaitMsec = cfg.RetryableHTTPClientConfig.RetryConfig.MinWaitMsec
+	}
+	if cfg.BlobConfig.MaxWaitMsec == 0 {
+		cfg.BlobConfig.MaxWaitMsec = cfg.RetryableHTTPClientConfig.RetryConfig.MaxWaitMsec
+	}
```

Comment:
Can we use (or make) a constructor for BlobConfig that does this, and call it from here?